lib/jekyll-mentions.rb

```diff
@@ -2,28 +2,36 @@
 require 'html/pipeline'
 
 module Jekyll
-  class Mentions < Jekyll::Generator
+  class Mentions < Jekyll::Converter
+    MENTIONABLE_EXTS = %w{.html .htm .md .markdown .mkdn .mkd .textile}
+
     safe true
+    priority :low
+
+    attr_reader :filter
 
     DEFAULT_URL = "https://github.com"
 
     def initialize(config = Hash.new)
-      @filter = HTML::Pipeline::MentionFilter.new(nil, {:base_url => base_url(config['jekyll-mentions']) })
+      @filter = HTML::Pipeline::MentionFilter.new(nil, {
+        :base_url => base_url(config['jekyll-mentions'])
+      })
     end
 
-    def generate(site)
-      site.pages.each { |page| mentionify page if html_page?(page) }
-      site.posts.each { |post| mentionify post }
-      site.docs_to_write.each { |doc| mentionify doc }
+    def matches(ext)
+      MENTIONABLE_EXTS.include?(ext.downcase)
     end
 
-    def mentionify(page)
-      return unless page.content.include?('@')
-      page.content = @filter.mention_link_filter(page.content)
+    def output_ext(*)
+      ".html".freeze
     end
 
-    def html_page?(page)
-      page.html? || page.url.end_with?('/')
+    def convert(content)
```

Comment:
As Ben mentioned, this could receive the previous converter's output extension. It'd be html here in that case and we'd be home free.